ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -362,30 +362,30 @@ static char *pdo_pgsql_last_insert_id(pdo_dbh_t *dbh, const char *name, size_t *
 	pdo_pgsql_db_handle *H = (pdo_pgsql_db_handle *)dbh->driver_data;
 	char *id = NULL;
 
-	if (name == NULL) {
-		if (H->pgoid == InvalidOid) {
-			return NULL;
-		}
-		*len = spprintf(&id, 0, ZEND_LONG_FMT, (zend_long) H->pgoid);
+	PGresult *res;
+	ExecStatusType status;
+	const char *q[1];
+	q[0] = name;
+
+	if (PHP_PDO_PGSQL_LASTVAL_PG_VERSION <= PQserverVersion && name == NULL) {
+		res = PQexec(H->server, "SELECT LASTVAL()");
 	} else {
-		PGresult *res;
-		ExecStatusType status;
-		const char *q[1];
-		q[0] = name;
 		res = PQexecParams(H->server, "SELECT CURRVAL($1)", 1, NULL, q, NULL, NULL, 0);
-		status = PQresultStatus(res);
+	}
+	status = PQresultStatus(res);
 
-		if (res && (status == PGRES_TUPLES_OK)) {
-			id = estrdup((char *)PQgetvalue(res, 0, 0));
-			*len = PQgetlength(res, 0, 0);
-		} else {
-			pdo_pgsql_error(dbh, status, pdo_pgsql_sqlstate(res));
-		}
+	if (res && (status == PGRES_TUPLES_OK)) {
+		id = estrdup((char *)PQgetvalue(res, 0, 0));
+		*len = PQgetlength(res, 0, 0);
+	} else {
+		pdo_pgsql_error(dbh, status, pdo_pgsql_sqlstate(res));
+        *len = spprintf(&id, 0, ZEND_LONG_FMT, (zend_long) H->pgoid);
```

Comment:
The identation is crrect according to the blocks coded in the function. You can check that now. ;-)